### DISCUSSION TITLE
S3 time out on write files

### DISCUSSION THREAD

**Discussion by MatinZare** [seaweedfs/seaweedfs#6158] [General]
Hi,
EDIT : i'am  using 3.63
I'm currently running a SeaweedFS setup with around 10 volume servers tagged with SSD disks, and a filer using a replication setting of `001`. I'm encountering an issue when trying to write files (ranging from 1 byte to 1 MB) using the command` mcli cp` to a path like `myfiler/mybucket/myfolder/`. The command hangs and does not complete even after 36 minutes.

Interestingly, I'm able to write files to other directories like `myfiler/mybucket/` or `myfiler/mybucket/myfolder2/` without any issues. The `myfolder` directory contains approximately 2 million objects, and I'm using ScyllaDB as the filer store. I also noticed that the write requests are not appearing in the filer logs.

Additionally, when I run the `mcli cp` command, the CPU usage of the filer spikes in a pattern that looks similar to running an `ls` command on the `myfolder` directory.

Here is a snippet from the command log with mcli in debug mode:
```


Host: fff.com
User-Agent: MinIO (linux; amd64) minio-go/v7.0.70 mcli/RELEASE.2024-05-09T17-04-24Z
Accept-Encoding: zstd,gzip
Authorization: AWS4-HMAC-SHA256 Credential=**REDACTED**, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
X-Amz-Content-Sha256: UNSIGNED-PAYLOAD
X-Amz-Date: 20241022T163055Z

mcli: <DEBUG> HTTP/1.1 200 OK
Content-Length: 169
Accept-Ranges: bytes
Content-Type: application/xml
Date: Tue, 22 Oct 2024 16:30:56 GMT
Server: SeaweedFS S3
X-Amz-Request-Id: 1729614656240624504

mcli: <DEBUG> Response Time: 661.528108ms

mcli: <DEBUG> GET /cdn/?object-lock= HTTP/1.1
Host: fff.com
User-Agent: MinIO (linux; amd64) minio-go/v7.0.70 mcli/RELEASE.2024-05-09T17-04-24Z
Accept-Encoding: zstd,gzip
Authorization: AWS4-HMAC-SHA256 Credential=**REDACTED**, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
X-Amz-Date: 20241022T163056Z

mcli: <DEBUG> HTTP/1.1 200 OK
Content-Length: 2594136
Accept-Ranges: bytes
Content-Type: application/xml
Date: Tue, 22 Oct 2024 16:30:56 GMT
Server: SeaweedFS S3
X-Amz-Request-Id: 1729614656524352450
```

Any insights into why this might be happening or suggestions on how to resolve the issue would be greatly appreciated!

**Comment by chrislusf** [comment]
checked FAQ #1?  https://github.com/seaweedfs/seaweedfs/wiki/FAQ#how-many-volumes-do-i-need

**Reply by MatinZare** [reply]
Yes, there is enough free disk space to create volumes with the desired replication placement. As I mentioned earlier, all the paths are within the same bucket, so they share the same collection in SeaweedFS. I have also preallocated volumes to ensure capacity. I verified this using the `weed shell` and the `volume.list` command, which shows that I have volumes tath are not  at their maximum capacity or readonly .

**Reply by chrislusf** [reply]
1. upgrade
2. take a thread dump of the s3 server by `kill -3`, if you think `weed s3` is stuck.
